### PR TITLE
Cases ETL, incremental & reporting

### PR DESCRIPTION
**Pull request by labkey-bpatel**
#### Rationale
Implement Cases ETL which includes 'Clinical Treatment' event.

#### Changes
* Add cases dataset, etl setup, reporting
* Add subqueries for audit event and staff to be reused for other ETLs
* Update drug ETL to use subqueries for audit event and staff

### CODE REVIEW

File: nirc_ehr/resources/queries/dbo/q_cases.sql [hunk 1]
```diff
@@ -1,20 +1,34 @@
...
+WHERE evtEvtGrp.EVENT_GROUP_ID = 37   -- 37 Clinical Treatment
+
   AND anmEvt.CREATED_DATETIME < now() -- there are rows in ANIMAL_EVENT table with future dates
---
-GROUP BY anmEvt.ANIMAL_EVENT_ID, anm.ANIMAL_ID_NUMBER, anmEvt.EVENT_DATETIME, adt.CHANGE_DATETIME, anmEvt.CREATED_DATETIME,
-anmCmt.TEXT, anmEvt.EVENT_ID.NAME
---ORDER BY adt.AUDIT_ID ASC
\ No newline at end of file
+
+-- Joining with audit_trail generates duplicate rows (ex. having multiple audit records for a given ANIMAL_EVENT_ID), hence the 'group by'
```
Comment (labkey-jeckels) [inline nirc_ehr/resources/queries/dbo/q_cases.sql]: There are multiple rows with the same adt.CHANGE_DATETIME value? That seems odd. Seems like we'd want the MAX(adt.CHANGE_DATETIME) and should remove it from the grouping, but I'm not looking at the raw data either

File: nirc_ehr/resources/queries/dbo/auditEvent.sql [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+         SELECT substring(adt.PRIMARY_KEY_VALUES, length('ANIMAL_EVENT_ID = '))             AS objectid,
+                CAST(adt.CHANGE_DATETIME AS TIMESTAMP)                                      AS modified,
+                adt.REFERENCE,
+                SUBSTR(adt.reference,
+                       instr(adt.reference, 'Event:') + length('Event:'))                   AS event_partial,
+                SUBSTR(adt.reference,
+                       instr(adt.reference, 'Event Datetime:') + length('Event Datetime:')) AS event_datetime
+         FROM AUDIT_TRAIL adt
+         WHERE adt.PRIMARY_KEY_VALUES LIKE '%ANIMAL_EVENT_ID%'
+           AND adt.COLUMN_NAME = 'DELETE') auditSubQuery
```
Comment (labkey-jeckels) [inline nirc_ehr/resources/queries/dbo/auditEvent.sql]: This is just pulling rows for deletes? If so, seems like we should name this query auditDeleteEvent or something like that.

**Review by labkey-jeckels — CHANGES_REQUESTED** [review]
A couple of suggestions or questions. Let me know if I'm mixed up.

File: nirc_ehr/resources/queries/dbo/q_cases.sql (hunk 1 above)
Comment (labkey-bpatel) [reply nirc_ehr/resources/queries/dbo/q_cases.sql]: "There are multiple rows with the same adt.CHANGE_DATETIME value?" - correct! Its on different columns, however. Perhaps both the columns are being changed at the same time but generating 2 rows in the audit_trail table.
In any case, now updated to use max.

File: nirc_ehr/resources/queries/dbo/auditEvent.sql (hunk 2 above)
Comment (labkey-bpatel) [reply nirc_ehr/resources/queries/dbo/auditEvent.sql]: done

File: nirc_ehr/resources/queries/dbo/q_cases.sql (hunk 1 above)
Comment (labkey-jeckels) [reply nirc_ehr/resources/queries/dbo/q_cases.sql]: Ah, I see. Yes, I bet a single user action is touching two columns, and generating two audit records. We don't need to copy the same row twice, so adjusting the group by (like you did already) is the right approach.

**Review by labkey-jeckels — APPROVED** [review]

**Pull request merged**